tools/gfx/cuda/render-cuda.cpp

```diff
@@ -222,16 +225,32 @@ class TextureCUDAResource : public TextureResource
     CUmipmappedArray m_cudaMipMappedArray = CUmipmappedArray();
 };
 
-class CUDAResourceView : public ResourceView
+class CUDAResourceView : public IResourceView, public RefObject
```

Comment:
Is the inheritance from `RefObject` on this stuff just for the convenience of not having to hand-write the reference-counting?

It seems like we eventually want to have some kind of `ComponentImpl<IWhatever>` base type that can be inherited from that handles all the boring bits of COM if possible. One of the key benefits of using that idiom would be that you don't end up with multiple inheritance and thus with every object containing multiple virtual function table pointers.

(The base type could even be `ComponentImpl<DerivedType, IWhatever>` if that gets rid of more implementation cruft)